homeassistant/components/watchyourlan/sensor.py

```diff
@@ -0,0 +1,149 @@
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import CONNECTION_NETWORK_MAC, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .coordinator import WatchYourLANUpdateCoordinator
+
+# Define entity descriptions for each sensor type
+ENTITY_DESCRIPTIONS = [
+    SensorEntityDescription(
+        key="online_status",
+        name="Online Status",
+        icon="mdi:lan-connect",
+    ),
+    SensorEntityDescription(
+        key="ip_address",
+        name="IP Address",
+        icon="mdi:ip-network",
+    ),
+    SensorEntityDescription(
+        key="mac_address",
+        name="MAC Address",
+        icon="mdi:lan",
+    ),
+    SensorEntityDescription(
+        key="iface",
+        name="Network Interface",
+        icon="mdi:ethernet",
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the WatchYourLAN sensors."""
+    coordinator: WatchYourLANUpdateCoordinator = entry.runtime_data
+
+    entities: list[SensorEntity] = [WatchYourLANDeviceCountSensor(coordinator)] + [
+        WatchYourLANGenericSensor(coordinator, device, description)
+        for device in coordinator.data
+        for description in ENTITY_DESCRIPTIONS
+    ]
+
+    async_add_entities(entities)
+
+
+class WatchYourLANGenericSensor(
+    CoordinatorEntity[WatchYourLANUpdateCoordinator], SensorEntity
+):
+    """Generic WatchYourLAN sensor for different data points."""
+
+    def __init__(
+        self,
+        coordinator: WatchYourLANUpdateCoordinator,
+        device: dict[str, Any],
+        description: SensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self.device = device
+        self.entity_description = description
+        self._attr_unique_id = f"{self.device.get('ID')}_{description.key}"
+
+    @property
+    def native_value(self) -> str:
+        """Return the native value of the sensor based on its description."""
+        if self.entity_description.key == "online_status":
+            return "Online" if self.device.get("Now", 0) == 1 else "Offline"
+        return self.device.get(self._get_device_field_for_key(), "Unknown")
+
+    def _get_device_field_for_key(self) -> str:
+        """Map description key to the appropriate device field."""
+        field_mapping = {
+            "online_status": "Now",
+            "ip_address": "IP",
+            "mac_address": "Mac",
+            "iface": "Iface",
+        }
+        return field_mapping.get(self.entity_description.key, "")
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return device info for the sensor."""
+        return DeviceInfo(
+            connections={
+                (CONNECTION_NETWORK_MAC, self.device.get("Mac", "00:00:00:00:00:00"))
```

Comment:
Didn't you say you always had a mac?

Reply:
We do/should, but I don't always trust APIs :) I figured it was just safer this way, instead of throwing some IndexError or something.

Reply:
If the API docs say that you have an mac address, just go for it. Once people report it doesn't work like expected, we can always implement a fix. I'd rather do this than have code that gives the impression it will be optional